Optimal Strategy for a game (Set -1)/optimal_strategy_for_a_game_set_1.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+OPTIMAL STRATEGY FOR A GAME SET - 1
+Given Problem Statement :
+             Problem statement: Consider a row of n coins of values v1 . . . vn, where n is even.
+             We play a game against an opponent by alternating turns. In each turn,
+             a player selects either the first or last coin from the row,
+             removes it from the row permanently, and receives the value of the coin.
+             Determine the maximum possible amount of money we can definitely win if we move first.
+       Note - Note: The opponent is as clever as the user.
+*/
+
+
+
+
+
+// C++ program to find out maximum value from a given sequence of coins
+  #include <bits/stdc++.h>
+  using namespace std;
+
+// Returns optimal value possible that a player can
+// collect from an array of coins of size n. Note
+// than n must be even
+int optimalStrategyOfGame(int* arr, int n)
+{
+    // Create a table to store solutions of subproblems
+    int table[n][n];
+
+    // Fill table using above recursive formula. Note
+    // that the table is filled in diagonal fashion from diagonal elements to table[0][n-1] which is the result.
+    for (int gap = 0; gap < n; ++gap) {
+        for (int i = 0, j = gap; j < n; ++i, ++j) {
+
+            // Here x is value of F(i+2, j), y is F(i+1, j-1) and
```

Comment:
keep space before and after arithmetic operators, also make this as a multi line commend